Doc/library/os.path.rst

```diff
@@ -306,17 +306,18 @@ the :mod:`glob` module.)
 
 .. function:: join(path, *paths)
 
-   Join one or more path components intelligently.  The return value is the
-   concatenation of *path* and any members of *\*paths* with exactly one
+   Join one or more path segments intelligently.  The return value is the
+   concatenation of *path* and all members of *\*paths*, with exactly one
    directory separator following each non-empty part except the last, meaning
    that the result will only end in a separator if the last part is empty.  If
```

Comment:
Going to change this as well, since `os.path.join("a", "b" + os.sep)` ends with a separator
```suggestion
   that the result will end in a separator if the last part is empty.  If
```